File: src/main/java/tomeko/hymod/utils/Waypoint.java
package tomeko.hymod.utils;

//? if = 1.8.9 {
import net.minecraft.util.BlockPos;
//?} else {
/*import net.minecraft.core.BlockPos;
*///?}

import java.awt.*;

public class Waypoint {
    public BlockPos pos;
    public Color color;
    public String text;
    public float boxOpacity;
    public float beamOpacity;
    public boolean renderText;
    public boolean renderDistance;
    public int tickTime;

    public Waypoint(
            BlockPos pos,
            Color color,
            String text,
            float boxOpacity,
            float beamOpacity,
            boolean renderText,
            boolean renderDistance,
            int tickTime
    ) {
        this.pos = pos;
        this.color = color;
        this.text = text;
        this.boxOpacity = boxOpacity;
        this.beamOpacity = beamOpacity;
        this.renderText = renderText;
        this.renderDistance = renderDistance;
        this.tickTime = tickTime;
    }
}

File: src/main/kotlin/tomeko/hymod/utils/Waypoint.kt
package tomeko.hymod.utils

//? if = 1.8.9 {
/*import cc.polyfrost.oneconfig.config.core.OneColor
import net.minecraft.util.BlockPos

*///?} else {
import net.minecraft.core.BlockPos
import org.polyfrost.compose.render.PolyColor
//?}

class Waypoint(
    var pos: BlockPos,
    var boxColor:
    //? if = 1.8.9 {
    /*OneColor,
    *///?} else {
    PolyColor,
    //?}
    var beamColor:
    //? if = 1.8.9 {
    /*OneColor,
    *///?} else {
    PolyColor,
    //?}
    var owner: String,
    var renderOwner: Boolean,
    var ownerColor:
    //? if = 1.8.9 {
    /*OneColor,
    *///?} else {
    PolyColor,
    //?}
    var text: String,
    var renderText: Boolean,
    var textColor:
    //? if = 1.8.9 {
    /*OneColor,
    *///?} else {
    PolyColor,
    //?}
    var renderDistance: Boolean,
    var distanceTextColor:
    //? if = 1.8.9 {
    /*OneColor,
    *///?} else {
    PolyColor,
    //?}
    var tickTime: Int
)